backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapper.java

```diff
@@ -0,0 +1,277 @@
+            materialItemStock.isBlocked(), materialItemStock.getLocationBpna());
+        var allocatedStocksList = new ArrayList<AllocatedStock>();
+        allocatedStocksList.add(allocatedStock);
+        position.setAllocatedStocks(allocatedStocksList);
+        return samm;
+    }
+
+    public ItemStockSAMM toItemStockSamm(ProductItemStock productItemStock) {
+        ItemStockSAMM samm = new ItemStockSAMM();
+        samm.setDirection(DirectionCharacteristic.OUTBOUND);
+        samm.setPositions(new ArrayList<>());
+        samm.setMaterialGlobalAssetId(productItemStock.getMaterial().getMaterialNumberCx());
+        samm.setMaterialNumberSupplier(productItemStock.getMaterial().getOwnMaterialNumber());
+        samm.setMaterialNumberCustomer(mprService.find(productItemStock.getMaterial(),
+            productItemStock.getPartner()).getPartnerMaterialNumber());
+        var posList = new ArrayList<Position>();
+        samm.setPositions(posList);
+        Position position = new Position();
+        if (productItemStock.getCustomerOrderId() != null || productItemStock.getCustomerOrderPositionId() != null
+            || productItemStock.getSupplierOrderId() != null) {
+            OrderPositionReference opr = new OrderPositionReference(productItemStock.getSupplierOrderId(),
+                productItemStock.getCustomerOrderId(), productItemStock.getCustomerOrderPositionId());
+            position.setOrderPositionReference(opr);
+        }
+        ItemQuantityEntity itemQuantityEntity = new ItemQuantityEntity(productItemStock.getQuantity(),
+            productItemStock.getMeasurementUnit());
+        AllocatedStock allocatedStock = new AllocatedStock(itemQuantityEntity, productItemStock.getLocationBpns(),
+            productItemStock.isBlocked(), productItemStock.getLocationBpna());
+        var allocatedStocksList = new ArrayList<AllocatedStock>();
+        allocatedStocksList.add(allocatedStock);
+        position.setAllocatedStocks(allocatedStocksList);
+        return samm;
+    }
+
+    public List<ReportedProductItemStock> sammToReportedProductItemStock(ItemStockSAMM samm, Partner partner) {
+        String matNbrCustomer = samm.getMaterialNumberCustomer();
+        String matNbrSupplier = samm.getMaterialNumberSupplier(); // should be ownMaterialNumber
+        String matNbrCatenaX = samm.getMaterialGlobalAssetId();
+        ArrayList<ReportedProductItemStock> outputList = new ArrayList<>();
+        if(samm.getDirection() != DirectionCharacteristic.INBOUND) {
+            log.warn("Direction should be INBOUND, aborting");
+            return outputList;
+        }
+        Material material = null;
+        // Use CatenaXNbr
+        if (matNbrCatenaX != null) {
+            material = materialService.findByMaterialNumberCx(matNbrCatenaX);
+            if (material != null) {
+                if (!material.getOwnMaterialNumber().equals(matNbrSupplier)) {
```

Comment:
Here or in next line, there seems to be a typo:
- if checks own (customer material number) against supplier material number
- log message indicates cx number and ownMaterialNumber

I assume the check should be between the customer material numbers to validate that our internal representation matches the external one. Is this right?

Reply:
See above. That method produces a ReportedProductItemStock. That means, it will be used on the suppliers side.

Reply:
confirmed